props-mongodb/src/main/java/sh/props/mongodb/MongoDbStore.java

```diff
@@ -188,6 +207,16 @@ static boolean isShardedCluster(MongoClient client) {
     return configDB.getCollection("shards").countDocuments() > 0;
   }
 
+  /**
+   * Initializes a {@link MongoClient} from the specified connection string.
+   *
+   * @param connectionString the MongoDB connection string
+   * @return an initialized connection to a valid cluster
+   */
+  private static MongoClient initClient(String connectionString) {
```

Comment:
calling methods that can be overridden in a constructor, means subclasses can impact the base class.  Instead, I provided a separate constructor that accepts a `MongoClient`